corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js

```diff
@@ -53,23 +53,29 @@ hqDefine("cloudcare/js/formplayer/menus/api", function () {
                                 menus.fetch($.extend(true, {}, options, { data: newOptionsData }));
                             }, gettext('Waiting for server progress'));
                         } else if (_.has(response, 'exception')) {
-                            FormplayerFrontend.trigger('clearProgress');
-                            FormplayerFrontend.trigger(
-                                'showError',
-                                response.exception,
-                                response.type === 'html'
-                            );
-
-                            var currentUrl = FormplayerFrontend.getCurrentRoute();
-                            if (FormplayerFrontend.lastError === currentUrl) {
-                                FormplayerFrontend.lastError = null;
-                                FormplayerFrontend.trigger('navigateHome');
+                            if (params.isClickableIcon &&
+                                response.exception.includes('Could not find case with ID')) {
```

Comment:
we should not depend on the exception text matching here.

Reply:
That's fair. I can change this. At the moment the error response from FP looks like this `"exception": "404 NOT_FOUND \"Could not find case with ID 471dbec6-3d5b-4d4b-835c-a1586da76702\"",
    "status": "error",`
So the value of `status` isn't `404`, which is what I initially expected. I could pull `404` from the `exception` or look at a way on FP side to have the status come back as `404`.

Reply:
Ahh I see, I think what we need is for FP to return 404, otherwise we will need to resort to previous method of using a response flag on FP instead to tell Web Apps the non existance of the case.